### PR TITLE
statistics: add serializer to generate the report csv file

### PR DESCRIPTION
**Pull request by vgranata**
* Create csv file for statistics reports.
* Transform data from list to table for reports with 2 distributions.

The test should be added after rebase of the PR for the permissions, without this PR the test fails.

Co-Authored-by: Valeria Granata <valeria@chaw.com>

## Why are you opening this PR?
https://github.com/rero/rero-ils/issues/3044

### CODE REVIEW

File: rero_ils/modules/stats/serializers.py [hunk 1]
```diff
@@ -75,6 +137,38 @@ def _format_csv(self, records):
                 value = StatCSVSerializer.sort_dict_by_key(value)[1]
                 writer.writerow(value)
                 yield line.read()
+        elif record['metadata'].get('type') == StatType.REPORT:
+            # statistics of type report
+            indicator = record['metadata']['config']['category']['indicator']
+            distributions = indicator.get('distributions')
```
Comment (zannkukai) [inline rero_ils/modules/stats/serializers.py:143]: use `distributions = indicator.get('distributions', [])` and later you could remove ... (see next comment)

File: rero_ils/modules/stats/serializers.py [hunk 2]
```diff
@@ -75,6 +137,38 @@ def _format_csv(self, records):
...
+            results = True
+
+            # case report with no query results
+            if len(record['metadata']['values']) == 1\
+               and record['metadata']['values'][0][indicator['type']] == 0:
+                results = False
+
+            # case report with 2 distributions
+            if distributions and len(distributions) == 2 and results:
```
Comment (zannkukai) [inline rero_ils/modules/stats/serializers.py:152]: ... you could remove `if distributions` condition :: `if len(distributions) == 2 and results:`

File: rero_ils/modules/stats/serializers.py [hunk 3]
```diff
@@ -75,6 +137,38 @@ def _format_csv(self, records):
...
+                # transform list into table
+                processed_data = self._create_table(record)
```
Comment (zannkukai) [inline rero_ils/modules/stats/serializers.py:154]: remove this line and use `for row_data in self._create_table(record):`

File: rero_ils/modules/stats/serializers.py [hunk 4]
```diff
@@ -42,6 +44,66 @@ class StatCSVSerializer(CSVSerializer):
...
     ]
 
+    def _create_table(self, record):
+        """Create table for reports with 2 distributions.
+
+        :param record: statistics report record
+        :returns: formatted data table
+        """
+        data = record['metadata']['values']
+        org_pid = data[0].get('org_pid')
```
[the remark this answers is not in the document]
Comment (vgranata) [reply rero_ils/modules/stats/serializers.py:56]: If the report has no results it shows 0 values:
https://github.com/rero/rero-ils/blob/US-statistics/rero_ils/modules/stats/api.py#L1211

**Review by zannkukai — APPROVED** [review]
again : missing "uni-test"

**Pull request merged**